### ISSUE TITLE
 INTERNAL: Internal error Rst Stream above okhttp 1.47.1 version

### ISSUE THREAD

**Issue by polingsky** [grpc/grpc-java#11051]
**What version of gRPC-Java are you using?**
1.47.1

**What is your environment?**
for Android, grpc verison v1.47.1
Tested with Android 6 to Android 14

app client ManagedChannel = OkHttpChannelBuilder.forAddress(TLS_HOST, TLS_PORT)
.overrideAuthority(TLS_HOST)
.sslSocketFactory(socketFactory)
.maxInboundMetadataSize(Integer.MAX_VALUE)
.maxInboundMessageSize(20 * 1024 * 1024)
.idleTimeout(2, TimeUnit.MINUTES)
.build();

**Server side**
.net 2.61.0

**What did you expect to see?**
I expect to be able to reuse the same ManagedChannel for each grpc call.
But it will somtimes fail and trigger INTERNAL: Internal error Rst Stream above okhttp 1.46.1 version (1.47.1 to 1.62.2).
This error not happend below okhttp 1.46.1 version (1.35.0 to 1.46.1)

**What did you see instead?**
INTERNAL: Internal error Rst Stream

**On Android client side**
INTERNAL: Internal error
Rst Stream|Status{code=INTERNAL, description=Internal error
Rst Stream, cause=null}

**On .net server side**
No error

**Comment by ejona86** [comment]
There's not much to go on here. None of the changes between 1.46 and 1.47 to okhttp/ and core/ jump out as interesting. HTTP/2 doesn't propagate much information for RST_STREAM(INTERNAL), so there's really no details to investigate on client-side. We really need more information from the server-side, or an approximate reproduction. I do expect there are very few sources of error that would cause a RST_STREAM(INTERNAL) in .net. Would you mind opening an issue on the .Net side (and CC me)? They may be able to help narrow it down.

**Comment by polingsky** [comment]
@ejona86 hello sir, thanks your reply
I post the issue to .Net project
https://github.com/grpc/grpc-dotnet/issues/2399

But I dont know how to CC to you?

**Comment by ejona86** [comment]
Thanks, that works. I had meant you could just mention me (`@ejona86`) on that issue, which auto-subscribes me. But I've subscribed to the issue now, so things are fine.

**Comment by ejona86** [comment]
Since this really needs more investigation from .Net before we can do anything, I'm going to close this for now since we can't do anything more. If you get additional info, comment and we can reopen, or file a new issue.
